joomla/joomla-cms
# Modal com_contact: improve Select & Edit Contact

Improve modals to select and to edit a contact.
**Note: to be testing on staging.**
#### Summary of Changes
- When Bootstrap modal loads an iframe, BS tooltips with placement top are truncated at the top border of the iframe. (no auto placement in BS2). This PR sets the tooltip placement to bottom to fix this issue.
- Remove inline CSS
- Content is embedded in a container div (using already existing container-popup class for modal.php)
- Add empty lines to improve the readability
- Add viewport dimensions (new <code>3.6.0-dev</code>)
- Contact edit modal is simplified (load edit layout, so no dupplicated code)
- Fix Edit button when current associated contact in data is not the selected contact
- Control if modal, in edit layout
- Add <code>Save</code> (keep modal open), <code>Save & Close</code> and <code>Close</code> button to the footer of the modal
- Fix Edit modal not opening as a modal (missing bootstrap.renderModal for Edit modal)
- Fix consistency of fieldsets <code>$this->ignore_fieldsets</code>, and add missing tabs, and remove dupplicated fieldset metadata (as using now correct edit view inside the modal)
- Consistency of language string <code>COM_CONTACT_EDIT_CONTACT</code> with other components (com_content, com_categories...) : "_Edit Contact_" and not "_Contact_".
#### Testing Instructions (Staging + Multilanguages enabled!)
- Go to Components > Contacts > open a contact, and go to <code>Associations</code> tab to select associated contact to open the modal.
- Verify tooltip placement is bottom, and not truncated
- Verify the modal size now is proportionnal to viewport (modal is 80/100th viewport width, and modal-body is 70/100th viewport height)
- Select an associated contact and save the contact to show edit button
- Open Edit modal and play with it ;-)
- Do the same in a menu item type "Single contact", for Select and Edit contact.

Same rendering as for Categories in PR #10441 (screenshots)
